src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -108,38 +108,45 @@ public void Start()
             stopping = false;
             pipelineInvocationTasks = new ConcurrentDictionary<Task, Task>();
 
-            
-            for(var i = 0; i < numberOfClients; i++)
+            var exceptions = new ConcurrentQueue<Exception>();
+            Parallel.For(0, numberOfClients, i =>
```

Comment:
Can't do this as the interfaces IManageMessageSenderLifeCycle and IManageMessageReceiverLifeCycle are public it would need to change to return a task instead, so need to wait until our next major